controller/admin_controller.php

```diff
@@ -0,0 +1,296 @@
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\controller;
+
+/**
+ * Admin controller
+ */
+class admin_controller implements admin_interface
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\language\language */
+	protected $language;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var array */
+	protected $cfg_array = array();
+
+	/** @var array */
+	protected $errors = array();
+
+	/** @var array */
+	protected $new_config = array();
+
+	/** @var string */
+	protected $phpbb_root_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var string */
+	public $u_action;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\config\config              $config               Config object
+	 * @param \phpbb\db\driver\driver_interface $db                   Database object
+	 * @param \phpbb\language\language          $language             Language object
+	 * @param \phpbb\log\log                    $log                  Log object
+	 * @param \phpbb\request\request            $request              Request object
+	 * @param \phpbb\template\template          $template             Template object
+	 * @param \phpbb\user                       $user                 User object
+	 * @param string                            $root_path            phpBB root path
+	 * @param string                            $php_ext              php_ext
+	 * @access public
+	 */
+	public function __construct(\phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\language\language $language, \phpbb\log\log $log, \phpbb\request\request $request, \phpbb\template\template $template, \phpbb\user $user, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function display_options($id, $mode)
+	{
+		// Create a form key for preventing CSRF attacks
+		add_form_key('acp_phpbb_ideas_settings');
+
+		$this->new_config = $this->config;
+		$this->cfg_array = ($this->request->is_set('config')) ? $this->request->variable('config', array('' => ''), true) : $this->new_config;
+
+		// Check if selected user exists
+		$this->check_ideas_topics_poster();
+
+		// Set Ideas forum  options and registered usergroup forum permissions
+		$this->set_ideas_forum_options($id, $mode);
+
+		// Do not write values if there are errors
+		$num_errors = sizeof($this->errors);
+		if ($num_errors)
+		{
+			$errors = implode('<br />', $this->errors);
```

Comment:
this->errors probably won't be needed when the set_ideas_forum_options method has its own logging/messaging and erroring. set_ideas_forum_options should not be sharing anything with any other method. So this could all be simplified more too where you probably shouldn't need `num_errors` and `errors`. There should only really be `errors`.